### PR TITLE
har converter: Correlate variables in recordings (just redirects for now)

### PR DESCRIPTION
**Pull request by cyberw**
Detect redirects in recordings and follow the redirects at runtime instead of following a hard-coded redirect. Also fix a bug with GET requests (http.get() does not take a body parameter)

### CODE REVIEW

File: converter/har/converter.go [hunk 1]
```diff
@@ -136,6 +153,15 @@ func Convert(h HAR, includeCodeCheck bool, returnOnFailedCheck bool, batchTime u
 						}
 					}
 				}
+
+				if e.Response.Headers != nil {
+					for _, header := range e.Response.Headers {
+						if header.Name == "Location" {
+							fmt.Fprintf(w, "\t\tredirectUrl = res.headers.Location;\n")
+							recordedRedirectUrl = header.Value
```
Comment (robingustafsson) [inline converter/har/converter.go]: We could add a `break` here to stop the for-loop early.

File: converter/har/converter.go [hunk 2]
```diff
@@ -55,7 +55,9 @@ func Convert(h HAR, includeCodeCheck bool, returnOnFailedCheck bool, batchTime u
 		fmt.Fprintf(w, "// %v\n", h.Log.Comment)
 	}
 
-	fmt.Fprint(w, "\n")
+	// recordings include redirections as separate requests, and we dont want to trigger them twice
+	fmt.Fprint(w, "\nexport let options = { maxRedirects: 0 };\n\n")
```
Comment (robingustafsson) [inline converter/har/converter.go]: Is it always the case with HAR files that redirects are included as separate requests? If not then we should perhaps make this a CLI flag as well.

**Comment by robingustafsson** [comment]
@cyberw Nice, will be interesting to see how far we can take the correlation in the future.

File: converter/har/converter.go (hunk 2 above)
Comment (cyberw) [reply converter/har/converter.go]: The HAR spec doesnt mention anything about redirects (except as part of responses) which would indicate that redirects are always represented as a separate request, but I'll double check using a browser based recorder (my proxy based recorder did generate separate requests)

File: converter/har/converter.go (hunk 1 above)
Comment (cyberw) [reply converter/har/converter.go]: sure, but why add the code complexity? this isnt "runtime" code so a few extra loops wont hurt anyone.. also, maybe we'll want to act on additional headers in the future (maybe unlikely but who knows...)

File: converter/har/converter.go (hunk 2 above)
Comment (cyberw) [reply converter/har/converter.go]: Confirmed. See attached file if you dont believe me ;)



[www.google.se.har.json.txt](https://github.com/loadimpact/k6/files/1699736/www.google.se.har.json.txt)

Comment (robingustafsson) [reply converter/har/converter.go]: Alright, cool, I believe you. Just checking :)

File: converter/har/converter.go (hunk 1 above)
Comment (robingustafsson) [reply converter/har/converter.go]: Sure, but that's in a possible future :) It's not a big thing at all, just something I noted as a possible improvement for the present, remove unnecessary cpu cycles if possible.

File: converter/har/converter.go (hunk 2 above)
Comment (cyberw) [reply converter/har/converter.go]: :) not a bad question, I hadnt really double checked..

File: converter/har/converter.go (hunk 1 above)
Comment (cyberw) [reply converter/har/converter.go]: IMO its premature optimization, but I'll add it :)

**Comment by cyberw** [comment]
Cool. My next step will be correlating repeated REST calls (where each post is inherently similar to the previous response)

**Comment by cyberw** [comment]
I think the CI failure is not my fault...

**Comment by robingustafsson** [comment]
No, that's one of the flakiness issues (https://github.com/loadimpact/k6/issues/469) with the engine tests we need to fix. I'll rebuild, and then merge.

**Comment by cyberw** [comment]
Cool, thanks!

**Pull request merged**